tests/test_indexer.py

```diff
@@ -195,6 +196,25 @@ def test_indexed_file_access_error(self):
                                                          excluded_files=[inaccesssible_filename.replace(".", "_")])
 
     def test_subscription_notification_successful(self):
+        PostTestHandler.verify_payloads = True
```

Comment:
There's a lot of code duplication among these test cases.  For instance, `test_subscription_notification_successful` and `test_unsigned_ subscription_notification_successful` only differ in two places.  Please consider simplifying.

Also, `PostTestHandler.verify_payloads` is global state that needs to be unset if the test fails for some reason.

Reply:
I have tried to make these into one test with no code duplication, but because `self.verify_notification` assumes that only one notification is ever active at a given time, I'm unable to do so.

Reply:
I've fixed `PostTestHandler.reset()` to reset verify_payloads instead of the other variable (which was unused, so that was a bug).

Reply:
Why not have:

```
    def test_subscription_notification_successful(self):
        self._test_subscription_notification_successful_helper(
            True,
            dict(hmac_secret_key=PostTestHandler.hmac_secret_key, hmac_key_id="test"),
        )

    def test_unsigned_subscription_notification_successful(self):
        self._test_subscription_notification_successful_helper(
            False,
            dict(),
        )

    def _test_subscription_notification_successful_helper(self, verify_payloads, subscribe_kwargs):
        PostTestHandler.verify_payloads = verify_payloads
        bundle_key = self.load_test_data_bundle_for_path("fixtures/smartseq2/paired_ends")
        sample_event = self.create_sample_bundle_created_event(bundle_key)
        self.process_new_indexable_object(sample_event, logger)

        ElasticsearchClient.get(logger).indices.create(self.subscription_index_name)
        subscription_id = self.subscribe_for_notification(smartseq2_paired_ends_query,
                                                          f"http://{HTTPInfo.address}:{HTTPInfo.port}",
                                                          **subscribe_kwargs)

        bundle_key = self.load_test_data_bundle_for_path("fixtures/smartseq2/paired_ends")
        sample_event = self.create_sample_bundle_created_event(bundle_key)
        self.process_new_indexable_object(sample_event, logger)
        prefix, _, bundle_id = bundle_key.partition("/")
        self.verify_notification(subscription_id, smartseq2_paired_ends_query, bundle_id)
```

Reply:
To address your code duplication concern, I added a utility function to delete a subscription, and unified these tests into one test that duplicates less code.